apps/src/Sound.js

```diff
@@ -339,22 +340,97 @@ Sound.prototype.getPlayableFile = function() {
   return false;
 };
 
-Sound.prototype.preload = function() {
-  var file = this.getPlayableFile();
+/**
+ * Checks if bytes were provided and we are able to play them after decoding.
+ */
+Sound.prototype.getPlayableBytes = function() {
+  try {
+    if (!window.Audio) {
+      return false;
+    }
+
+    let audioTest = new window.Audio();
+    if (
+      this.config.hasOwnProperty('bytes') &&
+      audioTest.canPlayType('audio/mp3')
+    ) {
+      return this.config.bytes;
+    }
+  } catch (e) {
+    console.warn('No bytes provided or mp3 is not supported');
+  }
+
+  return false;
+};
+
+Sound.prototype.preloadFile = function() {
+  const file = this.getPlayableFile();
   if (!file) {
     return;
   }
 
   if (!this.config.forceHTML5 && window.AudioContext && this.audioContext) {
     var self = this;
-    this.preloadViaWebAudio(file, function(buffer) {
+    this.preloadViaWebAudio(file, buffer => {
       self.reusableBuffer = buffer;
     });
     return;
   }
 
   if (window.Audio) {
-    var audioElement = new window.Audio(file);
+    let audioElement = new window.Audio(file);
+    if (!audioElement || !audioElement.play) {
```

Comment:
One last comment: Looks like lines from here down are the same for `preloadFile` and `preloadBytes`
It may make sense to factor those into shared function. i.e.

```
preloadFile = function() {
  ...
  if (window.Audio) {
    let audioElement = new window.Audio(file);
    preload(audioElement);
  }
}